packages/frontend/src/components/SavedDashboards/AddTilesToDashboardModal.tsx

```diff
@@ -196,11 +226,51 @@ const AddTilesToDashboardModal: FC<AddTilesToDashboardModalProps> = ({
 
     const currentSpace = spaces?.find((s) => s.uuid === tile?.props.spaceUuid);
 
+    const isDashboardSelectItemDisabled = useCallback(
+        (dashboard: DashboardBasicDetailsWithTileTypes) => {
+            switch (dashboardTileType) {
+                case DashboardTileTypes.SAVED_CHART:
+                case DashboardTileTypes.SQL_CHART:
+                    return dashboard.tileTypes.includes(
+                        DashboardTileTypes.SEMANTIC_VIEWER_CHART,
+                    );
+                case DashboardTileTypes.SEMANTIC_VIEWER_CHART:
+                    return (
+                        dashboard.tileTypes.includes(
+                            DashboardTileTypes.SAVED_CHART,
+                        ) ||
+                        dashboard.tileTypes.includes(
+                            DashboardTileTypes.SQL_CHART,
+                        )
+                    );
+                case DashboardTileTypes.LOOM:
+                case DashboardTileTypes.MARKDOWN:
+                    return false;
+                default:
+                    return assertUnreachable(
+                        dashboardTileType,
+                        `Unknown tile type: ${dashboardTileType}`,
+                    );
+            }
+        },
+        [dashboardTileType],
+    );
+
+    const dashboardSelectItems = useMemo(() => {
+        return (
+            dashboards?.map<ItemProps>((d) => ({
+                value: d.uuid,
+                label: d.name,
+                group: spaces?.find((s) => s.uuid === d.spaceUuid)?.name,
+                disabled: isDashboardSelectItemDisabled(d),
+                spaceUuid: d.spaceUuid, // ? Adding spaceUuid here for simplicity of selecting the default value in the select
+            })) ?? []
+        );
+    }, [dashboards, isDashboardSelectItemDisabled, spaces]);
+
     const form = useForm({
         initialValues: {
```

Comment:
if it helps and this is a mantine form - you can initialize the form later https://mantine.dev/form/recipes/#set-initial-values-with-async-request